cloudevents/formats/avro-turbo-format.md

```diff
@@ -0,0 +1,109 @@
+# Avro Turbo Event Format for CloudEvents - Version 1.0.3-wip
+
+## Abstract
+
+The Avro Turbo Format for CloudEvents defines how events are expressed in
+the [Avro 1.9.0 Specification][avro-spec].
+
+This differs from the [Avro format](avro-format.md) in that:
+
+- It is optimized for performance, preferring a more compact representation.
+- It only supports spec version 1.0 (any changes to spec version requires changes to the Avro schema,
+  which changes the fingerprint, breaking compatibility).
+- It does not natively support JSON (JSON can be straight-forwardly serialized 
+  to bytes and this was therefore not considered necessary).
+
+## Table of Contents
+
+1. [Introduction](#1-introduction)
+2. [Attributes](#2-attributes)
+3. [Transport](#3-transport)
+4. [Examples](#4-examples)
+
+## 1. Introduction
+
+[CloudEvents][ce] is a standardized and protocol-agnostic definition of the
+structure and metadata description of events. This specification defines how the
+CloudEvents are to be represented as [Avro 1.9.0][avro-primitives].
+
+The [Attributes](#2-attributes) section describes the naming conventions and
+data type mappings for CloudEvents attributes for use as Avro message
+properties.
+
+This specification does not define an envelope format. The Avro type system's
+intent is primarily to provide a consistent type system for Avro itself and not
+for message payloads.
+
+The Avro event format does not currently define a batch mode format.
+
+### 1.1. Conformance
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC2119][rfc2119].
+
+## 2. Attributes
+
+This section defines how CloudEvents attributes are mapped to the Avro
+type-system. This specification explicitly maps each attribute.
+
+### 2.1 Type System Mapping
+
+The CloudEvents type system MUST be mapped to Avro types as follows.
+
+| CloudEvents   | Avro                                                                   |
+|---------------|------------------------------------------------------------------------|
+| Boolean       | [boolean][avro-primitives]                                             |
+| Integer       | [int][avro-primitives]                                                 |
+| String        | [string][avro-primitives]                                              |
+| Binary        | [bytes][avro-primitives]                                               |
+| URI           | [string][avro-primitives] following [RFC 3986 §4.3][rfc3986-section43] |
+| URI-reference | [string][avro-primitives] following [RFC 3986 §4.1][rfc3986-section41] |
+| Timestamp     | [long][avro-primitives]  using `timestamp-micros` logical type         |
+
+Extension specifications MAY define secondary mapping rules for the values of
```

Comment:
I've never understood this in any of our formats. Maybe @JemDay or @duglin could clarify... otherwise I'm happy to take an action item to remove it everywhere possible :)

Reply:
it just means that CE attribute values could also appear in other places. I believe https://github.com/cloudevents/spec/blob/main/cloudevents/extensions/distributed-tracing.md is an example of that. The traceparent value appears as both as CE attribute and an HTTP header. We didn't want to imply that people must replace some other spec's mechanism with CE's, but rather they can co-exist.